plugins/modules/mount.py

```diff
@@ -659,6 +710,25 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     return mounts
 
 
+def _is_same_mount_src(module, args, mntinfo_file="/proc/self/mountinfo"):
+    """Return True if the mounted fs on mountpoint is the same source than src"""
```

Comment:
This function is used for the `ephemeral` *state*: if this function returns False, that means that the provided mount point has already a volume mounted on it, and it is NOT the volume given in *src*. This leads to a module failure, to avoid an unwanted unmount.